integrationTests/multiShard/smartContract/testInitilalizer.go

```diff
@@ -0,0 +1,888 @@
+		pkBuff, _ := n.pk.ToByteArray()
+
+		fmt.Printf("Shard ID: %v, sk: %s, pk: %s\n",
+			n.shardId,
+			hex.EncodeToString(skBuff),
+			hex.EncodeToString(pkBuff),
+		)
+		_ = n.node.Start()
+		_ = n.node.P2PBootstrap()
+	}
+}
+
+func createNodes(
+	numOfShards int,
+	nodesPerShard int,
+	serviceID string,
+) []*testNode {
+
+	//first node generated will have is pk belonging to firstSkShardId
+	numMetaChainNodes := 1
+	nodes := make([]*testNode, int(numOfShards)*nodesPerShard+numMetaChainNodes)
+
+	idx := 0
+	for shardId := 0; shardId < numOfShards; shardId++ {
+		for j := 0; j < nodesPerShard; j++ {
+			testNode := &testNode{
+				dPool:   createTestShardDataPool(),
+				shardId: uint32(shardId),
+			}
+
+			shardCoordinator, _ := sharding.NewMultiShardCoordinator(uint32(numOfShards), uint32(shardId))
+			accntAdapter := createAccountsDB()
+			n, mes, sk, resFinder, blkProcessor, txProcessor, transactionCoordinator, scrForwarder, blkc, store := createNetNode(
+				testNode.dPool,
+				accntAdapter,
+				shardCoordinator,
+				testNode.shardId,
+				serviceID,
+			)
+			_ = n.CreateShardedStores()
+
+			testNode.node = n
+			testNode.sk = sk
+			testNode.messenger = mes
+			testNode.pk = sk.GeneratePublic()
+			testNode.resFinder = resFinder
+			testNode.accntState = accntAdapter
+			testNode.blkProcessor = blkProcessor
+			testNode.txProcessor = txProcessor
+			testNode.scrForwarder = scrForwarder
+			testNode.blkc = blkc
+			testNode.store = store
+			testNode.txCoordinator = transactionCoordinator
+			testNode.dPool.Headers().RegisterHandler(func(key []byte) {
+				atomic.AddInt32(&testNode.headersRecv, 1)
+				testNode.mutHeaders.Lock()
+				testNode.headersHashes = append(testNode.headersHashes, key)
+				header, _ := testNode.dPool.Headers().Peek(key)
+				testNode.headers = append(testNode.headers, header.(data.HeaderHandler))
+				testNode.mutHeaders.Unlock()
+			})
+			testNode.dPool.MiniBlocks().RegisterHandler(func(key []byte) {
+				atomic.AddInt32(&testNode.miniblocksRecv, 1)
+				testNode.mutMiniblocks.Lock()
+				testNode.miniblocksHashes = append(testNode.miniblocksHashes, key)
+				miniblock, _ := testNode.dPool.MiniBlocks().Peek(key)
+				testNode.miniblocks = append(testNode.miniblocks, miniblock.(*dataBlock.MiniBlock))
+				testNode.mutMiniblocks.Unlock()
+			})
+			testNode.dPool.MetaBlocks().RegisterHandler(func(key []byte) {
+				fmt.Printf("Got metachain header: %v\n", base64.StdEncoding.EncodeToString(key))
+				atomic.AddInt32(&testNode.metachainHdrRecv, 1)
+			})
+			testNode.dPool.Transactions().RegisterHandler(func(key []byte) {
+				atomic.AddInt32(&testNode.txsRecv, 1)
+			})
+			testNode.broadcastMessenger, _ = sposFactory.GetBroadcastMessenger(
+				testMarshalizer,
+				mes,
+				shardCoordinator,
+				sk,
+				&singlesig.SchnorrSigner{},
+			)
+
+			nodes[idx] = testNode
+			idx++
+		}
+	}
+
+	shardCoordinatorMeta, _ := sharding.NewMultiShardCoordinator(uint32(numOfShards), sharding.MetachainShardId)
```

Comment:
use one one shard coordinator. Not needed to recreate it for every shard and meta.

Reply:
Shard coordinators have a selfId, this is why it's recreated for each. Some tests might call selfId

Reply:
true. :)